packages/react-native-bpk-theming/src/BpkThemeProvider.js

```diff
@@ -22,6 +22,16 @@ import createWithTheme from './create-with-theme';
 
 const BpkThemeProvider = ThemeProvider;
 
+// TODO (BPK-3639): Upgrade theming package and remove this
+const cwm = ThemeProvider.prototype.componentWillMount;
```

Comment:
Could you clarify what this logic is doing here?

Reply:
I'm just renaming `componentWillMount` and `componentWillReceiveProps` to `UNSAFE_componentWillMount` and `UNSAFE_componentWillReceiveProps` to remove the deprecation warnings. Once we update the theming package we can remove this, but because we don't expose it directly to consumers we can do this little hack to remove the warning for now and avoid polluting the app with even more warnings.

Reply:
I am wondering if we should be hacking this instead of just moving to the correct replacements and fixes for this?

Reply:
The only thing we can do to remove this is upgrading the theming lib, and it's a breaking change, so I'd rather not do that with the RN upgrade.